src/yews/train/functional.py

```diff
@@ -12,6 +12,7 @@ def get_torch_device():
 
 def model_on_device(model, device):
     return torch.nn.DataParallel(model.to(device))
+    #return torch.nn.DataParallel(model,device_ids=[0]).to("cuda")
```

Comment:
I understand that you want to select which GPU to use. However, that is precisely the reason why I passed in `device` as an argument. You may consider using something below:

```python
device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
model_on_device(model, device)
```
In general, you want to control the selection of GPU in your main script instead of inside the trainer class. Thus, it is better to be passed in as an argument.

Reply:
Resolve. Please check it again. Thanks.